### PR TITLE
Improve code actions reported for missing associated field

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
$subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/4198

## Examples
Screenshot 2023-03-14 at 13 29 03

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the specification~
- [ ] ~Updated the changelog~
- [x] Added tests
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-optional-type.bal [hunk 1]
```diff
@@ -51,13 +50,14 @@ type Building3 record {|
     string country;
     string postalCode;
     Workspace3? workspace2;
+	Workspace3 workspace3;
```
Comment (daneshk) [inline compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-optional-type.bal:53]:
```suggestion
    Workspace3? workspace2;
    Workspace3 workspace3;
```

File: compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-type-1.bal [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    string workspaceType;
+|};
+
+type Building1 record {|
+    readonly string buildingCode;
+    string city;
+    string state;
+    string country;
+    string postalCode;
+	Workspace1? workspace1;
```
Comment (daneshk) [inline compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-type-1.bal:23]:
```suggestion
    string postalCode;
    Workspace1? workspace1;
```

File: compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-type-2.bal [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    string workspaceType;
+|};
+
+type Building1 record {|
+    readonly string buildingCode;
+    string city;
+    string state;
+    string country;
+    string postalCode;
+	Workspace1[] workspace1;
```
Comment (daneshk) [inline compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-type-2.bal:23]:
```suggestion
    string postalCode;
    Workspace1[] workspace1;
```

File: compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-optional-type.bal (hunk 1 above)
Comment (niveathika) [reply compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-optional-type.bal:53]: This is formatted due to the added tabspace. When inserting the row, I am adding a tabspace before hand. Should this be changed to 4 spaces?

File: compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-type-2.bal (hunk 3 above)
Comment (niveathika) [reply compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-type-2.bal:23]: This seems to be a bug in Github PR preview. Locally this is generated correctly.

File: compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-type-1.bal (hunk 2 above)
Comment (niveathika) [reply compiler-plugin-test/src/test/resources/codeaction/mandatory-relation-entity-type-1.bal:23]: This seems to be a bug in Github PR preview. Locally this is generated correctly.

**Pull request merged**